### PR TITLE
Add table access rules to FileBasedSystemAccessControl

### PR DESCRIPTION
**Pull request by haldes**
Currently FileBasedSystemAccessControl control has catalogRules, queryAccessRules, impersonationRules, principalUserMatchRules and schemaAccessRules. 
The proposal here is to add table access rules to it. This will give more granular access controls over the datasets.

At high level idea is to implement/enhance the below methods of FileBasedSystemAccessControl

```
1. checkCanDropTable 				- Allow if user has OWNERSHIP access to table
2. checkCanRenameTable 			- Allow if user has OWNERSHIP access to table
3. checkCanGrantTablePrivilege		- Allow if user has OWNERSHIP access to table
4. checkCanRevokeTablePrivilege	        - Allow if user has OWNERSHIP access to table
5. checkCanShowCreateTable		- Allow if user has OWNERSHIP access to table
6. checkCanAddColumn				- Allow if user has OWNERSHIP access to table
7. checkCanDropColumn			- Allow if user has OWNERSHIP access to table
8. checkCanRenameColumn			- Allow if user has OWNERSHIP access to table
9. checkCanSetTableComment		- Allow if user has OWNERSHIP access to table
10. checkCanShowColumns  			- Allow if user has any the SELECT, INSERT, DELETE, OWNERSHIP, GRANT_SELECT access to table
11. filterColumns					- Allow if user has any the SELECT, INSERT, DELETE, OWNERSHIP, GRANT_SELECT access to table
12. checkCanSelectFromColumns          - Allow if user has SELECT access to table
13. checkCanInsertIntoTable			- Allow if user has INSERT access to table
14. checkCanDeleteFromTable		- Allow if user has DELETE access to table
```
Please find below the slack discussion on the same
https://prestosql.slack.com/archives/CFLB9AMBN/p1588921035123100?thread_ts=1507847605.000156&cid=CFLB9AMBN

@kokosing @findepi

### CODE REVIEW

**Comment by kokosing** [comment]
```
3. checkCanGrantTablePrivilege		- Allow if user has OWNERSHIP access to table
4. checkCanRevokeTablePrivilege	        - Allow if user has OWNERSHIP access to table
```

How does it work with file based access control? User cannot GRANT or REVOKE anything because configuration is static (controlled by the file content, which could be modified externally)? Do you modify the file (haven't reviewed the code yet)?

**Comment by haldes** [comment]
@kokosing - yes you can modify the access control file and can assign TablePreviledges like SELECT, INSERT, DELETE, OWNERSHIP, GRANT_SELECT  against a user

File: presto-plugin-toolkit/src/main/java/io/prestosql/plugin/base/security/FileBasedSystemAccessControl.java [hunk 1]
```diff
@@ -617,4 +683,34 @@ private boolean isSchemaOwner(SystemSecurityContext context, String schemaName)
...
+        return checkTablePermission(context, tableName, privileges -> privileges.containsAll(ImmutableSet.copyOf(requiredPrivileges)));
+    }
+
+    private boolean checkTablePermission(SystemSecurityContext context, SchemaTableName tableName, Predicate<Set<TableAccessControlRule.TablePrivilege>> checkPrivileges)
+    {
+        if (tableRules.isEmpty()) {
+            return true;
+        }
+
+        if (INFORMATION_SCHEMA_NAME.equals(tableName.getSchemaName())) {
```
Comment (kokosing) [inline presto-plugin-toolkit/src/main/java/io/prestosql/plugin/base/security/FileBasedSystemAccessControl.java:703]: I don't think anyone should be allowed to modify `INFORMATION_SCHEMA_NAME` tables. But I see it is how it is implemented in `FileBasedAccessControl`.

Comment (haldes) [reply presto-plugin-toolkit/src/main/java/io/prestosql/plugin/base/security/FileBasedSystemAccessControl.java:703]: @findepi added to #1559

**Comment by haldes** [comment]
@kokosing @electrum @findepi 
Is there any thing else i need to do for this PR?

**Pull request merged**